code/modules/client/client procs.dm

```diff
@@ -427,4 +489,111 @@
 /client/proc/destroy_UI()
 	if(UI)
 		qdel(UI)
-		UI = null
\ No newline at end of file
+		UI = null
+
+//Uses a couple different services
+/client/proc/update_ip_reputation()
+	var/list/scores = list("GII" = ipr_getipintel())
+	if(config.ipqualityscore_apikey)
+		scores["IPQS"] = ipr_ipqualityscore()
+	/* Can add other systems if desirable
+	if(config.blah_apikey)
+		scores["blah"] = some_proc()
+	*/
+
+	var/log_output = "IP Reputation [key] from [address]"
+	var/worst = 0
+
+	for(var/service in scores)
+		var/score = scores[service]
+		if(score > worst)
+			worst = score
+		log_output += " - [service] ([num2text(score)])"
+
+	log_admin(log_output)
+	ip_reputation = worst
+	return TRUE
+
+//Service returns a single float in html body
+/client/proc/ipr_getipintel()
+	if(!config.ipr_email)
+		return -1
+
+	var/request = "http://check.getipintel.net/check.php?ip=[address]&contact=[config.ipr_email]"
+	var/http[] = world.Export(request)
+
+	if(!http || !islist(http)) //If we couldn't check, the service might be down, fail-safe.
+		log_admin("Couldn't connect to getipintel.net to check [address] for [key]")
+		return -1
+
+	//429 is rate limit exceeded
+	if(text2num(http["STATUS"]) == 429)
+		log_and_message_admins("getipintel.net reports HTTP status 429. IP reputation checking is now disabled. If you see this, let a developer know.")
```

Comment:
BTW. Can't that message spam to admins too much (in case of service failure)?
May be it should be debug log?